peas/sensors.py

```diff
@@ -25,45 +21,25 @@ def __init__(self, auto_detect=False, *args, **kwargs):
 
         assert 'environment' in self.config
         assert type(self.config['environment']) is dict, \
-            self.logger.warning("Environment config variable not set correctly. No sensors listed")
+            self.logger.warning('Environment config variable not set correctly. No sensors listed')
 
         self.db = None
         self.messaging = None
 
         # Store each serial reader
         self.serial_readers = dict()
 
-        if auto_detect:
+        if auto_detect or self.config['environment'].get('auto_detect', False):
+            self.logger.debug('Performing auto-detect')
             for port_num in range(9):
```

Comment:
We can also grab connected ports and just check those: https://pythonhosted.org/pyserial/tools.html?highlight=comport#serial.tools.list_ports.comports. We could then check vendor id and product id. The arduino uno even reports a `serial_number` correctly although I don't think the micros do.

That could probably help clean up some of the `_auto_detect_port` below.

Reply:
Submitted #284 for this.